src/lib.rs

```diff
@@ -1063,22 +1062,9 @@ impl fmt::Debug for Command {
 
 /// Moves `Fd` out of non-blocking mode.
 #[cfg(unix)]
-fn blocking_fd(fd: std::os::unix::io::RawFd) -> io::Result<()> {
-    // Helper macro to execute a system call that returns an `io::Result`.
-    macro_rules! syscall {
-        ($fn:ident ( $($arg:expr),* $(,)? ) ) => {{
-            let res = unsafe { libc::$fn($($arg, )*) };
-            if res == -1 {
-                return Err(std::io::Error::last_os_error());
-            } else {
-                res
-            }
-        }};
-    }
-
-    let res = syscall!(fcntl(fd, libc::F_GETFL));
-    syscall!(fcntl(fd, libc::F_SETFL, res & !libc::O_NONBLOCK));
-
+fn blocking_fd(fd: rustix::fd::BorrowedFd<'_>) -> io::Result<()> {
+    let flags = rustix::fs::fcntl_getfl(fd)?;
```

Comment:
Looking at [`libstd`](https://doc.rust-lang.org/src/std/sys/unix/fd.rs.html#252-275), on Linux it uses [`FIONBIO`](https://docs.rs/rustix/latest/rustix/io/fn.ioctl_fionbio.html) instead of this strategy. We might want to try that instead.

Reply:
That behavior in the standard library seems to have been introduced in https://github.com/rust-lang/rust/commit/efeb42be2837842d1beb47b51bb693c7474aba3d.

I wondered why it is linux-only, since FIONBIO is available on several other unix-like systems. After a little looking into it, on `std` it seems mainly because it could not find docs on other operating systems (https://github.com/rust-lang/rust/pull/39514#issuecomment-277429921), and also [this comment in libuv](https://github.com/libuv/libuv/blob/e9d91fccfc3e5ff772d5da90e1c4a24061198ca0/src/unix/poll.c#L78-L80) indicates that there are cases where it is not supported, at least on BSD. Also, [this commit in mio](https://github.com/tokio-rs/mio/commit/0db49f6d5caf54b12176821363d154384357e70a) indicates that it is not working as we expected on illumos.

So, for now, it seems to make sense to do it only on linux (and android).